js_modules/dagster-ui/packages/ui-core/src/assets/catalog/AssetSelectionSummaryListItemV2.tsx

```diff
@@ -0,0 +1,125 @@
+import {
+  BodySmall,
+  Box,
+  Colors,
+  HorizontalControls,
+  ListItem,
+  MiddleTruncate,
+} from '@dagster-io/ui-components';
+import React, {useMemo} from 'react';
+import {Link} from 'react-router-dom';
+
+import {useSelectionHealthData} from './useSelectionHealthData';
+import {ViewType, getHealthStatuses, getThreadId, useAssetHealthStatuses} from './util';
+import {InsightsIcon, InsightsIconType} from '../../insights/InsightsIcon';
+import {numberFormatter} from '../../ui/formatters';
+import {AssetTableFragment} from '../types/AssetTableFragment.types';
+
+export const AssetSelectionSummaryListItemFromSelection = React.memo(
+  ({index, item}: {index: number; item: Extract<ViewType, {__typename: 'CatalogView'}>}) => {
+    const {liveDataByNode, assetCount, loading} = useSelectionHealthData({
+      selection: item.selection.querySelection ?? '',
+    });
+    useMemo(
+      () => getHealthStatuses({liveDataByNode, loading, assetCount}),
+      [liveDataByNode, loading, assetCount],
+    );
+    return (
+      <AssetSelectionSummaryListItemWithHealthStatus
+        index={index}
+        icon={<InsightsIcon name={item.icon as InsightsIconType} size={16} />}
+        label={item.name}
+        link={item.link}
+        loading={loading}
+        assetCount={assetCount}
+      />
+    );
+  },
+);
+
+interface AssetSelectionSummaryListItemProps {
+  index: number;
+  assets: AssetTableFragment[];
+  icon: React.ReactNode;
+  label: string;
+  link: string;
+  loading?: boolean;
+}
+
+export const AssetSelectionSummaryListItem = React.memo(
```

Comment:
I probably wouldn't bother `memo`ing these unless there's some performance bottleneck.

In particular because `icon` is a `ReactNode`, and unless you're memoizing that value too, it's probably going to be a brand new object every time (which is fine, btw) and cause the `memo` to be pointless.